crt_portal/cts_forms/models.py

```diff
@@ -81,59 +81,75 @@ class Report(models.Model):
 
     # Commercial or public space
     commercial_or_public_place = models.CharField(max_length=225, choices=COMMERCIAL_OR_PUBLIC_PLACE_CHOICES, null=True, default=None)
-    other_commercial_or_public_place = models.CharField(max_length=150, blank=True, null=True)
+    other_commercial_or_public_place = models.CharField(max_length=150, blank=True, null=True, default=None)
 
     # Education location
     public_or_private_school = models.CharField(max_length=100, null=True, choices=PUBLIC_OR_PRIVATE_SCHOOL_CHOICES, default=None)
 
     # Incident date
-    last_incident_year = models.PositiveIntegerField(MaxValueValidator(datetime.now().year, message="Date must not be in the future"))
+    last_incident_year = models.PositiveIntegerField(MaxValueValidator(datetime.now().year, message="Date must not be in the future"), null=True, blank=True)
     last_incident_day = models.PositiveIntegerField(MaxValueValidator(31, message='Day value too high'), null=True, blank=True)
-    last_incident_month = models.PositiveIntegerField(MaxValueValidator(12, message="Month must be 12 or less"))
+    last_incident_month = models.PositiveIntegerField(MaxValueValidator(12, message="Month must be 12 or less"), null=True, blank=True,)
 
     # Internal comments
     internal_comments = models.ManyToManyField(InternalHistory, blank=True)
 
+    # Metadata
+    create_date = models.DateTimeField(auto_now_add=True)
+    modified_date = models.DateTimeField(auto_now=True)
+    crt_reciept_year = models.PositiveIntegerField(MaxValueValidator(datetime.now().year), null=True, blank=True)
+    crt_reciept_day = models.PositiveIntegerField(MaxValueValidator(31), null=True, blank=True)
+    crt_reciept_month = models.PositiveIntegerField(MaxValueValidator(12), null=True, blank=True)
+    intake_format = models.CharField(max_length=100, null=True, default=None, choices=INTAKE_FORMAT_CHOICES)
+    author = models.CharField(max_length=1000, null=True, blank=True)
+
     @cached_property
     def last_incident_date(self):
-        day = self.last_incident_day or 1
-        return datetime(self.last_incident_year, self.last_incident_month, day)
-
-    ###############################################################
-    #   These fields have not been implemented in the form yet:   #
-    ###############################################################
-    # where form
-    public_or_private_facility = models.CharField(max_length=100, null=True, choices=PUBLIC_OR_PRIVATE_FACILITY_CHOICES, default=None)
-    public_or_private_healthcare = models.CharField(max_length=100, null=True, choices=PUBLIC_OR_PRIVATE_HEALTHCARE_CHOICES, default=None)
+        try:
+            day = self.last_incident_day or 1
+            date = datetime(self.last_incident_year, self.last_incident_month, day)
+        except ValueError:
+            date = None
+        return date
+
+    @cached_property
+    def crt_reciept_date(self):
```

Comment:
I don't think these functions, `last_incident_date` and `crt_reciept_date` were actually modified here but noting that they don't appear to be call anywhere in the current code base. Potential opportunity to remove them here.

Reply:
`last_incident_date` was required and now isn't required for the pro form. We will need that when we add date range sort on the incident date. I try to front-load model changes as much as possible, because I feel like doing a lot of little model changes across a few people can get harry fast. 

`crt_reciept_date` is a new optional field for the pro form, but there isn't currently a place for its display on the back end. The use case for this info is to figure out how much time it take DOJ to get the mail to CRT. A date filed can be queried for those kinds of questions in a way that is easier than querying the month day and year fields. So the data will be available and packaged for data calls even if there isn't a nice way to get the data now. I have the related day, month and year fields just to keep parody with how we are inputing and processing the date fields for the front end.